core/workflow/aggregatorrole.go

```diff
@@ -64,6 +65,42 @@ func (r *aggregatorRole) UnmarshalYAML(unmarshal func(interface{}) error) (err e
 	return
 }
 
+func (r *aggregatorRole) MarshalYAML() (interface{}, error) {
+	// Aux struct to cast Roles as an array
+	type rootWorkflow struct {
```

Comment:
If it's an auxiliary struct I'd rather it be called something other than `rootWorkflow`. The reason is that "the root role of the workflow tree" already has a specific meaning (see `load.go`).

Generally, a role that's at the root of a workflow is necessarily of type `aggregatorRole`, but `aggregatorRole`s can also occur elsewhere in a workflow. See https://github.com/AliceO2Group/ControlWorkflows/blob/master/workflows/readout-stfb-qc.yaml#L28 for an example of this.

Post-MVP, an effort to implement workflow composition i.e. attaching a converted DPL workflow onto an existing AliECS workflow essentially boils down to appending subworkflow structures onto the `roles` field of an existing `aggregatorRole`.

Reply:
Changed `rootWorkflow` to `auxAggregatorRole` (since this comprises `roleBase` values and an `aggregator`).